aopy/analysis.py

```diff
@@ -1543,3 +1543,63 @@ def interp_multichannel(x):
     x[nan_idx] = np.interp(idx,xp,fp)
 
     return x
+
+
+'''
+Behavioral metrics 
+'''
+
+def compute_path_length_per_trial(trajectory):
+    '''
+    This function calculates the path length by computing the distance from all points for a single trajectory. The input trajectry could be cursor or eye trajectory from a single trial. It returns a single value for path length.
+
+    Args:
+        trajectory (nt x 2): single trial trajectory, could be a cursor trajectory or eye trajectory
+
+    Returns:
+        path_length (float): length of the trajectory
+    '''
+    lengths = np.sqrt(np.sum(np.diff(trajectory, axis=0)**2, axis=1)) # compute the distance from all points in trajectory
+    path_length = np.sum(lengths)
+    return path_length
+
+
+def time_to_target(event_codes, event_times, target_codes=list(range(81, 89)) , go_cue_code=32 , reward_code=48, only_rewarded_trials=True, per_target_stats=False):
+    '''
+    This function calculates reach time to target only on rewarded trials given trial aligned event codes and event times See: :func:`aopy.preproc.base.get_trial_segments_and_times` .
+
+    Args:
+        event_codes (list) : trial aligned event codes
+        event_times (list) : trial aligned event times corresponding to the event codes. These event codes and event times could be the output of preproc.base.get_trial_segments_and_times()
+        target_codes (list) : list of event codes for cursor entering peripheral target 
+        go_cue_code (int) : event code for go cue 
+        reward_code (int) : event code for reward 
+        only_rewarded_trials (bool, optional): only include trials with a reward code
+        per_target_stats (bool, optional): use if you want to calculate reach time per target
+
+    Returns:
+        reach_times (list):
+        reach_times_per_target(list of lists) : optional if per_target_stats == True
+    '''
+    event_times = np.array(event_times)
+    event_codes = np.array(event_codes)
+    if only_rewarded_trials:
```

Comment:
I am not sure I understand why this is helpful. The rest of the calculation outside the if loop will break if event_times and event_codes are a list of lists that are not of the uniform shape.

Reply:
One way of including a more generic trial filter is to get it as a lambda function instead.

Reply:
To be more clear, the function breaks because of some unsuccessful trials, such as the ones with center hold penalty with have shorter event_codes and event_times length, in which case, the rest of the calculations won't work. the condition for time to target is basically that the cursor enters the peripheral target.
This is what the lambda function could allow the user to choose. I hope my point gets across.

Reply:
hmm i see what you mean.  is it always going to be the case that rewarded trials are the same length? maybe we should just compute the lengths of the trials one at a time to avoid this issue in all cases?

Reply:
yes, always rewarded trials will have same event code sequence and will be of same length. I think we can keep the fucntion as it was before and add a note saying this will only work for rewarded trials. 
I will soon be generalizing to unrewarded trials with full trajectories as well. Like I said before, I will treat that as an enhancement.

Reply:
i added another function called `calc_segment_duration` which does similar calculations but on any segment. i also found a bug in your code if there are more than 1 of the same target then the per target stats are wrong, they only take the first appearance of each target into account. i think you need to take a mean somewhere.

Reply:
Could you explain a use case or the need for another function here? The time_to_target() does what it says it does. It would not be possible for me to test edge case scenarios extensively. However, I know this function works with our beignet's data and is generic enough to be applied to any session or task. 
The event codes for cursor entering peripheral targets are supposed to be unique. This function works as long as the codes are unique. Taking means within function would change the statistics when we concatenate reach times for example, so I avoided that.